shared/slices/slice_generic_test.go

```diff
@@ -0,0 +1,93 @@
+package slices
+
+import (
+	"reflect"
+	"testing"
```

Comment:
For these functions can you add tests using different data types rather than only testing the library with `GenericItem` ?
We dont really know if this library would work with different data types because all test cases use `GenericItem`

Reply:
Yeah I would prefer there to be explicit declaration of types so something like 
`[]uint64{2,8,5}`
`[]float64{2.5,7.3,8.9}`
etc .

Reply:
Generics are not supported by Go and in order to make the function in above manner we need to maintain functions for each data type. @nisdas any other idea apart from this which I can explore.

Reply:
I know generics are not supported by go, currently this library only tests slices of `GenericItem`. The purpose of #1193 was for us to be able to give slices of a larger variety of types ex: []uint64 , []byte , etc . and get the result. 

But currently as how the library is I can only give slices of `GenericItem` to get it to work. If I try giving it slices of another type, it will not work as the compiler will report it as incompatible. It defeats the purpose of #1193 if I can only use it of the `GenericItem` type.

Reply:
@nisdas this is corrected.